diff --git a/app/buck2_client/src/commands/build/out.rs b/app/buck2_client/src/commands/build/out.rs
index c67d22bb47..dff3b22228 100644
--- a/app/buck2_client/src/commands/build/out.rs
+++ b/app/buck2_client/src/commands/build/out.rs
@@ -551,6 +551,7 @@ mod tests {
             let out = dir.path().join("sleep");

             let res = buck2_util::process::async_background_command("cp")
+                .arg("-L")
                 .arg(Path::new("/bin/sleep"))
                 .arg(&out)
                 .spawn()?
@@ -559,6 +560,15 @@ mod tests {

             assert!(res.success());

+            let res = buck2_util::process::async_background_command("chmod")
+                .arg("+wx")
+                .arg(&out)
+                .spawn()?
+                .wait()
+                .await?;
+
+            assert!(res.success());
+
             let mut proc = buck2_util::process::async_background_command(&out)
                 .arg("10000")
                 .kill_on_drop(true)
