diff --git a/compiler/rustc_codegen_ssa/src/errors.rs b/compiler/rustc_codegen_ssa/src/errors.rs
index cceda09c701..a8696bc1318 100644
--- a/compiler/rustc_codegen_ssa/src/errors.rs
+++ b/compiler/rustc_codegen_ssa/src/errors.rs
@@ -1,5 +1,7 @@
 //! Errors emitted by codegen_ssa

+#![allow(unused)]
+
 use std::borrow::Cow;
 use std::ffi::OsString;
 use std::io::Error;
diff --git a/compiler/rustc_codegen_ssa/src/target_features.rs b/compiler/rustc_codegen_ssa/src/target_features.rs
index 2572d216082..36227b2afee 100644
--- a/compiler/rustc_codegen_ssa/src/target_features.rs
+++ b/compiler/rustc_codegen_ssa/src/target_features.rs
@@ -229,7 +229,7 @@ fn parse_rust_feature_list<'a>(
 /// We do not have to worry about RUSTC_SPECIFIC_FEATURES here, those are handled elsewhere.
 pub fn cfg_target_feature<'a, const N: usize>(
     sess: &Session,
-    to_backend_features: impl Fn(&'a str) -> SmallVec<[&'a str; N]>,
+    _to_backend_features: impl Fn(&'a str) -> SmallVec<[&'a str; N]>,
     mut target_base_has_feature: impl FnMut(&str) -> bool,
 ) -> (Vec<Symbol>, Vec<Symbol>) {
     let known_features = sess.target.rust_target_features();
@@ -281,32 +281,7 @@ pub fn cfg_target_feature<'a, const N: usize>(
             // Check feature validity.
             let feature_state = known_features.iter().find(|&&(v, _, _)| v == base_feature);
             match feature_state {
-                None => {
-                    // This is definitely not a valid Rust feature name. Maybe it is a backend
-                    // feature name? If so, give a better error message.
-                    let rust_feature = known_features.iter().find_map(|&(rust_feature, _, _)| {
-                        let backend_features = to_backend_features(rust_feature);
-                        if backend_features.contains(&base_feature)
-                            && !backend_features.contains(&rust_feature)
-                        {
-                            Some(rust_feature)
-                        } else {
-                            None
-                        }
-                    });
-                    let unknown_feature = if let Some(rust_feature) = rust_feature {
-                        errors::UnknownCTargetFeature {
-                            feature: base_feature,
-                            rust_feature: errors::PossibleFeature::Some { rust_feature },
-                        }
-                    } else {
-                        errors::UnknownCTargetFeature {
-                            feature: base_feature,
-                            rust_feature: errors::PossibleFeature::None,
-                        }
-                    };
-                    sess.dcx().emit_warn(unknown_feature);
-                }
+                None => {}
                 Some((_, stability, _)) => {
                     if let Err(reason) = stability.toggle_allowed() {
                         sess.dcx().emit_warn(errors::ForbiddenCTargetFeature {
@@ -314,19 +289,6 @@ pub fn cfg_target_feature<'a, const N: usize>(
                             enabled: if enable { "enabled" } else { "disabled" },
                             reason,
                         });
-                    } else if stability.requires_nightly(/* in_cfg */ false).is_some() {
-                        // An unstable feature. Warn about using it. It makes little sense
-                        // to hard-error here since we just warn about fully unknown
-                        // features above.
-                        let note = if stability.is_cfg_stable_toggle_unstable() {
-                            "this feature is allowed in cfg but unstable otherwise"
-                        } else {
-                            "this feature is not stably supported"
-                        };
-                        sess.dcx().emit_warn(errors::UnstableCTargetFeature {
-                            feature: base_feature,
-                            note,
-                        });
                     }
                 }
             }
